lib/extensions/ar_taggable.rb

```diff
@@ -148,7 +148,8 @@ def is_vtagged_with?(tag, options = {})
         relationship = "self"
         macro = :has_one
       else
-        macro = subject.class.reflect_on_association(relationship.to_sym).macro
+        klass = subject.class
```

Comment:
I think we can use `reflection_with_virtual` what is basically same what you are doing now.

cc @kbrock